### PR TITLE
updated hist documentation

### PR DESCRIPTION
**Pull request by mitchnegus**
pandas.DataFrame.hist()/pandas.Series.hist() currently say that the `bins` keyword argument must be an integer, with a default value of 10. Since these methods are built off matplotlib, these actually seem to be more flexible (allowing sequences as well as integers). I don't know if there was a reason for this discrepancy, but if not, I've adjusted the pandas docstrings to reflect the matplotlib documentation.

[Existing pandas documentation](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.hist.html)
[Existing matplotlib documentation](https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.hist.html)

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -2156,10 +2156,18 @@ def hist_frame(data, column=None, by=None, grid=True, xlabelsize=None,
...
+        Number of histogram bins to be used. If an integer is given, bins + 1
+        bin edges are calculated and returned. If bins is a sequence, gives
+        bin edges, including left edge of first bin and right edge of last
+        bin. In this case, bins is returned unmodified.
     `**kwds` : other plotting keyword arguments
         To be passed to hist function
+
+    Notes
+    -----
+    See matplotlib documentation online for more on this
```
Comment (gfyoung) [inline pandas/plotting/_core.py]: Could you point to a specific `matplotlib` function(s) with a "See Also" section in the docstring?

Comment (TomAugspurger) [reply pandas/plotting/_core.py]: `matplotlib.axes.Axes.hist` *might* be turned into a link automatically, if you add it to "See Also".

**Review by TomAugspurger — COMMENTED** [review]
Could you also update `hist_frame`

**Comment by TomAugspurger** [comment]
@mitchnegus do you have a chance to update based on these comments?

**Comment by mitchnegus** [comment]
Yes, sorry, I got busy and had forgot about this. Thanks for the bump!

**Review by TomAugspurger — APPROVED** [review]
Thanks, this all looks good to me. Please ping if you see that all the CI tests have passed before we do.

**Comment by jreback** [comment]
thanks!

**Pull request merged**